client/app/routes/collection/add.js

```diff
@@ -207,28 +230,39 @@ export default Ember.Route.extend({
                 },
                 parameters: {
                     parameter: 'upload_section'
-                }
+                },
+                then: 'cdefb5db-2486-4466-8c1c-5d24ffd7e6ab'
             }, {
-                type: 'delete_widget',
+                id: 'cdefb5db-2486-4466-8c1c-5d24ffd7e6ab',
+                type: 'closeSection',
+                args: {
+                    sectionName: 'upload'
+                },
+                then: '7fb38183-d1f2-41a2-aef5-1bc99743762e'
+            }, {
+                id: '7fb38183-d1f2-41a2-aef5-1bc99743762e',
+                type: 'disableWidget',
                 parameters: {
                     widget_object: 'save_upload_section_widget'
                 },
-                output_parameter: 'null',
-                conitions: [{
-                    all: [{
-                        parameter: 'upload_section',
-                        state: 'saved',
-                    }]
-                }]
+                then: '6606b697-9a53-4fc7-aab6-898a2904c579'
```

Comment:
As discussed previously, `then` should be renamed to `callback`. If it gets run after the data is saved, `saveCallback` would be even clearer.